doxia-core/src/test/java/org/apache/maven/doxia/sink/impl/AbstractSinkTest.java

```diff
@@ -110,7 +110,7 @@ public final void testTestDocument() throws IOException {
             SinkTestDocument.generate(testSink);
         } finally {
             testSink.close();
-            IOUtil.close(writer);
+            IOUtils.close(writer);
```

Comment:
This one might be able to use try with resources.

Reply:
Looked at this briefly, but `Sink` does not implement `AutoCloseable`, which makes it slightly awkward, especially as some of the multiple `Sink#close()` implementations call out to `init()` for instance, so leaving out `testSink.close()` is also not an option. You'd then have to declare `Sink` outside the try with resources, and close that through finally, which could change the order of the `close` calls if I'm not mistaken. Any other suggestions on how to handle this here?

Reply:
https://issues.apache.org/jira/browse/DOXIA-698 but until then this probably works. Does closing the sink close the Writer? If so we don't need IOUtils at all here.

The other pre-existing issue that;s weird in this test is the isXmlSink() branch on line 106. I'm not sure what that's about.

Reply:
It's `Sink.close()` documented as that it should: